torch/distributed/tensor/_dtensor_spec.py

```diff
@@ -93,9 +93,37 @@ def __post_init__(self) -> None:
             self.placements = tuple(self.placements)
         if self.shard_order is None:
             # pyrefly: ignore [bad-assignment]
-            self.shard_order = DTensorSpec.compute_default_shard_order(self.placements)
+            if any(isinstance(p, _StridedShard) for p in self.placements):
+                self.shard_order = self._maybe_convert_StridedShard_to_shard_order(
```

Comment:
what should happen here if `_maybe_convert` refuses to convert?

Reply:
should we just replace this if/else block with `self.shard_order = _normalize_placements...`?  
and add a comment explaining that if `self.shard_order` is None, that means the placements contains un-normalizable SS?